Parser/Types/TypeAnalyze.cs

```diff
@@ -136,11 +141,22 @@ public override NType Visit (NUnary u)
    public override NType Visit (NIdentifier d) {
       if (mSymbols.Find (d.Name.Text) is NVarDecl v) 
          return d.Type = v.Type;
+      if (mSymbols.Find (d.Name.Text) is NConstDecl c)
+         return d.Type = c.Type;
       throw new ParseException (d.Name, "Unknown variable");
    }
 
    public override NType Visit (NFnCall f) {
-      throw new NotImplementedException ();
+      if (mSymbols.Find (f.Name.Text) is NFnDecl g) {
+         if (f.Params.Length != g.Params.Length) throw new ParseException (f.Name, "Parameters length is not equal");
+         Visit (f.Params);
+         for (int i = 0; i < g.Params.Length; i++) {
+            f.Params[i] = AddTypeCast (f.Name, f.Params[i], f.Type);

```

Comment:
Instead of f.Type you should use g.Params[i].Type. See method AddTypeCase for details.